File: Java/Library/Event.java
import java.time.LocalDateTime;

public class Event {
    private LocalDateTime datetime;
    private Long datetimeEpoch;
    private String type;
    private Double latitude;
    private Double longitude;
    private Double distance;
    private String description;
    private Double size;

    public LocalDateTime getDatetime() {
        return datetime;
    }
    public void setDatetime(LocalDateTime datetime) { this.datetime = datetime; }

    public Long getDatetimeEpoch() { return datetimeEpoch; }
    public void setDatetimeEpoch(Long datetimeEpoch) { this.datetimeEpoch = datetimeEpoch; }

    public String getType() { return type; }
    public void setType(String type) { this.type = type; }

    public Double getLatitude() { return latitude; }
    public void setLatitude(Double latitude) { this.latitude = latitude; }

    public Double getLongitude() { return longitude; }
    public void setLongitude(Double longitude) { this.longitude = longitude; }

    public Double getDistance() { return distance; }
    public void setDistance(Double distance) { this.distance = distance; }

    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }

    public Double getSize() { return size; }
    public void setSize(Double size) { this.size = size; }
}

File: Scala/Library/Event.scala
import java.time.LocalDateTime

class Event(
             private var _datetime: Option[LocalDateTime] = None,
             private var _datetimeEpoch: Option[Long] = None,
             private var _type: Option[String] = None,
             private var _latitude: Option[Double] = None,
             private var _longitude: Option[Double] = None,
             private var _distance: Option[Double] = None,
             private var _description: Option[String] = None,
             private var _size: Option[Double] = None
           ) {

  def datetime: Option[LocalDateTime] = _datetime
  def datetime_=(datetime: Option[String]): Unit = {
    _datetime = datetime.map(LocalDateTime.parse)
  }

  def datetimeEpoch: Option[Long] = _datetimeEpoch
  def datetimeEpoch_=(datetimeEpoch: Option[Long]): Unit = {
    _datetimeEpoch = datetimeEpoch
  }

  def `type`: Option[String] = _type
  def type_=(t: Option[String]): Unit = {
    _type = t
  }

  def latitude: Option[Double] = _latitude
  def latitude_=(latitude: Option[Double]): Unit = {
    _latitude = latitude
  }

  def longitude: Option[Double] = _longitude
  def longitude_=(longitude: Option[Double]): Unit = {
    _longitude = longitude
  }

  def distance: Option[Double] = _distance
  def distance_=(distance: Option[Double]): Unit = {
    _distance = distance
  }

  def description: Option[String] = _description
  def description_=(description: Option[String]): Unit = {
    _description = description
  }

  def size: Option[Double] = _size
  def size_=(size: Option[Double]): Unit = {
    _size = size
  }
}
